exporter/redis.go

```diff
@@ -52,6 +64,50 @@ func (e *Exporter) connectToRedis() (redis.Conn, error) {
 	return c, err
 }
 
+func (e *Exporter) connectToRedisCluster() (redis.Conn, error) {
+	uri := e.redisAddr
+	if strings.Contains(uri, "://") {
+		url, _ := url.Parse(uri)
+		if url.Port() == "" {
+			uri = url.Host + ":6379"
+		} else {
+			uri = url.Host
+		}
+	} else {
+		if frags := strings.Split(uri, ":"); len(frags) != 2 {
+			uri = uri + ":6379"
+		}
+	}
+
+	options, err := configureOptions(e, uri)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Creating cluster object")
+	cluster := redisc.Cluster{
```

Comment:
If the `connectToRedisCluster()` returns `c` object _(which is `redisc.RetryConn()`, should be deferred on the top level)_, how can we `defer cluster.Close()`? `cluster` will be created for each  call of  the `scrapeHost() -> connectToRedisCluster()` and I'm not sure about possible _"memory"_ or _"connection leak"_ without explicit closing.

I apologize for such non-obvious questions, but it really worries me from the side of the stability of this monitoring project, servicing the need of thousands of operations engineers.

Reply:
That's an interesting question.
I think you're right, you're supposed to call `cluster.Close()` when done to release all the cluster resources. And we don't make `cluster` available to the caller so it's not possible.

But

If you look at the code for Close() [here](https://github.com/mna/redisc/blob/v1.2.1/cluster.go#L397-L411) then you notice that it's only closing all the pools and I think we don't use pools as we use `Dial()`.

From [the redisc docs](https://pkg.go.dev/github.com/mna/redisc#hdr-Cluster): 
`The Dial method, on the other hand, guarantees that the returned connection will not be managed by a pool` 

I'm not loving that we rely on that but I think I'm ok with it.